MetaMask/metamask-extension
# fix: Import all scripts fix

## **Description**

Fixes https://github.com/MetaMask/metamask-extension/issues/24514.

The problem was that if the service worker is terminated, when app-init.js runs again when the service worker is to be restarted, `importAllScripts()` is not called. The solution is just to make sure it is called in the top-level scope, on every run of the script. This should be safe to do because of the `scriptsLoadInitiated` check already in the `importAllScripts` function.

The PR also uses the `serviceworker.state` property to ensure that we don't attempt to import scripts before installation. (That is explained further in the code comment added in this PR)

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24612?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24514

## **Manual testing steps**

1. Start MM MV3 build
2. Go to chrome://serviceworker-internals/?devtools
3. Stop Service Worker manually with the button
4. You should be able to open MetaMask

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
